tenable/ad/users/api.py

```diff
@@ -0,0 +1,319 @@
+        '''
+        Retrieve all users
+
+        Returns:
+            list:
+                The list of users objects
+
+        Examples:
+
+            >>> tad.users.list()
+        '''
+        return self._schema.load(self._get(), many=True)
+
+    def create(self,
+               name: str,
+               email: str,
+               password: str,
+               **kwargs) -> List[Dict]:
+        '''
+        Create users
+
+        Args:
+            name:
+                The name of new user.
+            email:
+                The email address of the user.
+            password:
+                The password for the new user.
+            surname:
+                The surname of new user.
+            department:
+                The department of user.
+            biography:
+                The biography of user.
+            active:
+                is the user active?
+            picture:
+                The list of picture numbers
+
+        Return:
+            list[dict]:
+                The created user objects
+
+        Example:
+            >>> tad.users.create(
+            ...     name='username',
+            ...     email='test@domain.com',
+            ...     password='user_password',
+            ...     active=True
+            ...     )
+        '''
+        payload = self._schema.dump(dict_merge({
+            'name': name,
+            'email': email,
+            'password': password
+        }, kwargs))
+
+        return self._schema.load(self._post(json=payload), many=True)
+
+    def info(self) -> Dict:
+        '''
+        gets user information
+
+        Return:
+            dict:
+                The user info object
+
+        Example:
+            >>> tad.users.info()
+        '''
+        schema = UserInfoSchema()
+        return schema.load(self._get(f'whoami'))
+
+    def details(self, user_id: str) -> Dict:
+        '''
+        Retrieves the details for a specific user
+
+        Args:
+            user_id:
+                The user instance identifier.
+
+        Returns:
+            dict:
+                the user object.
+
+        Examples:
+
+            >>> tad.users.details(1)
+        '''
+        return self._schema.load(self._get(f'{user_id}'))
+
+    def update(self,
+               user_id: str,
+               **kwargs) -> Dict:
+        '''
+        update an existing user
+
+        Args:
+            user_id:
+                The user instance identifier.
+            name:
+                The name of new user.
+            email:
+                The email address of the user.
+            password:
+                The password for the new user.
+            surname:
+                The surname of new user.
+            department:
+                The department of user.
+            biography:
+                The biography of user.
+            active:
+                is the user active?
+            picture:
+                The list of picture numbers
```

Comment:
The docstring should call these out more explicitly like so:

```python
            user_id (str):
                 The user instance identifier.
             name (optional, str):
                 The name of new user.
             email (optional, str):
                 The email address of the user.
             password (optional, str):
                 The password for the new user.
             surname (optional, str):
                 The surname of new user.
             department (optional, str):
                 The department of user.
             biography (optional, str):
                 The biography of user.
             active (optional, bool):
                 is the user active?
             picture (optional, List[int]):
                 The list of picture numbers
             locked_out (optional, bool):
                 is the user locked out?
```

Reply:
updated all methods docstrings with type